source/SIL.AppBuilder.Portal.Frontend/src/ui/components/sidebar/header.tsx

```diff
@@ -25,90 +29,80 @@ export interface IProps {
   currentOrganization: ResourceObject<ORGANIZATIONS_TYPE, OrganizationAttributes>;
 }
 
-class SidebarHeader extends React.Component<IProps & i18nProps> {
-  componentDidMount() {
-    this.defaultSelectSingleOrgMembership();
-  }
-
-  componentDidUpdate() {
-    this.defaultSelectSingleOrgMembership();
-  }
-
-  defaultSelectSingleOrgMembership() {
-    const { organizations, currentOrganizationId, setCurrentOrganizationId } = this.props;
-
-    if (organizations) {
-      if (!currentOrganizationId && organizations.length === 1) {
-        const id = idFromRecordIdentity(organizations[0] as any);
-        setCurrentOrganizationId(id, false);
-      }
+// export default compose(
+//   withRouter,
+//   withCurrentUserContext,
+//   withCurrentOrganization,
+//   withRelationships((props: ICurrentUserProps) => {
+//     const { currentUser } = props;
+
+//     return {
+//       organizations: [currentUser, 'organizationMemberships', 'organization'],
+//     };
+//   }),
+//   withTranslations
+// )(SidebarHeader);
+
+export default function SidebarHeader(props: IProps) {
+  const { toggleOrgSwitcher: toggle, closeSidebar, isOrgSwitcherActive } = props;
+  const { t } = useTranslations();
+  const {
+    currentOrganization,
+    organizationsAvailableToUser,
+    currentUserOrganizations,
+    currentOrganizationId,
+    setCurrentOrganizationId,
+  } = useCurrentOrganization();
+
+  const hasMoreThanOneOrg = organizationsAvailableToUser.length > 1;
+
+  useEffect(() => {
+    if (!currentOrganizationId && organizationsAvailableToUser.length === 1) {
+      const id = idFromRecordIdentity(organizationsAvailableToUser[0] as any);
+
+      setCurrentOrganizationId(id, { skipNav: true });
     }
-  }
+  }, [currentOrganizationId]);
```

Comment:
woo hooks 🥇

Reply:
woo!
Turns out just by re-writing some things in the new format, I can also get rid of a bunch of previous hacks